### PR TITLE
Switch to diesel-async

### PR DESCRIPTION
**Pull request by banool**
## Summary
There are two main reasons to do this:
- Better perf since we don't block waiting for I/O to happen (esp if I can get pipelining to work).
- Breaking the dep on libpq.

As you can see, at least without pipelining, this PR has no real effect on perf, but in neither direction, it's just the same.

Update: I think I got pipelining to work (at least, it should be working as per the docs), but the impact was negligible. Probably our workload just doesn't benefit from it, not too sure: https://github.com/aptos-labs/aptos-indexer-processors/pull/147.

### Performance testing
Using this config:
```
health_check_port: 8084
server_config:
  processor_config:
    type: default_processor
  postgres_connection_string: postgresql://postgres:@localhost:5432/default_processor
  indexer_grpc_data_service_address: https://grpc.testnet.aptoslabs.com
  auth_token: <token>
```

Note: Prior to these results I ran this once all the way up to txn 2.5 million to warm up the cache upstream (just in case).

Experiment results. What txn we got up to after 3 minutes (last_success_version in the DB):

Built from main
1. Run 1: 2261999
1. Run 2: 2162999
1. Run 3: 2091999
1. Average: 2172332

Built from this branch
1. Run 1: 2250999
1. Run 2: 2231999
1. Run 3: 2030999
1. Average: 2171332
  
I alternated runs, so I did run 1 for each, then run 2 for each, then run 3 for each.

I ran tests with degraded networking: 100ms latency on downlink and uplink, 1% packet loss, and 50mbps bandwidth down / 10mbps up. There was no noticeable difference between the two builds (we weren't I/O bound in the end).

I used this command. I made sure the binary was built first:
```
timeout 180 target/release/processor -c processor/parser.yaml
```

### CODE REVIEW

**Comment by banool** [comment]
The alternative would be to do this: https://github.com/sfackler/rust-postgres/issues/890.

**Comment by banool** [comment]
See the benchmarking from the Diesel team themselves: https://github.com/diesel-rs/metrics.

File: rust/processor/src/processors/token_v2_processor.rs [hunk 1]
```diff
@@ -77,19 +78,19 @@ fn insert_to_db_impl(
...
 ) -> Result<(), diesel::result::Error> {
-    insert_collections_v2(conn, collections_v2)?;
-    insert_token_datas_v2(conn, token_datas_v2)?;
-    insert_token_ownerships_v2(conn, token_ownerships_v2)?;
-    insert_current_collections_v2(conn, current_collections_v2)?;
-    insert_current_token_datas_v2(conn, current_token_datas_v2)?;
-    insert_current_token_ownerships_v2(conn, current_token_ownerships_v2)?;
-    insert_token_activities_v2(conn, token_activities_v2)?;
-    insert_current_token_v2_metadatas(conn, current_token_v2_metadata)?;
+    insert_collections_v2(conn, collections_v2).await?;
```
Comment (larry-aptos) [inline rust/processor/src/processors/token_v2_processor.rs:81]: qq, we're sequentially inserting data into multiple tables under one transaction, is it doable that we use `tokio::spawn` and `select_all` to parallelize the db operations?

Comment (larry-aptos) [reply rust/processor/src/processors/token_v2_processor.rs:81]: oh i see the pipelining part.

**Review by bowenyang007 — APPROVED** [review]

**Pull request merged**